### PR TITLE
Add User class and CRUD for Income tracking

### LINKED ISSUE #66: Create a user class
(contains user name, income, expenditure, savings,  etc)

### PR DESCRIPTION
**Pull request by khseah**
Fixes #66

- Add User class (which contains ExpenditureList, CreditCardList and IncomeList)

- Add CRUD for Income (similar to CRUD for expenditure and credit cards)

- Update help command to list help related to Income

- Add some JUnit tests to test CRUD functionalities of Income (didn't add alot of tests, so those who need bump up their LOC can add some more tests, specifically to test "update", "list" and "help" of Income, and negative test cases)

- Refactor some code and rephrase some Javadoc comments to increase consistency

### CODE REVIEW

File: src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java [hunk 1]
```diff
@@ -35,14 +35,18 @@
...
-        if (!(input.contains(startingFlag + " ") && input.contains(" " + endingFlag))) {
-            throw new MindMyMoneyException("You are missing a flag or lack the spacing between the flags!");
-        }
-        startingFlag = startingFlag + " ";
-        input = input.substring(input.indexOf(startingFlag) + startingFlag.length());
-        if (!endingFlag.equals("")) {
-            endingFlag = " " + endingFlag;
-            input = input.substring(0, input.indexOf(endingFlag));
+        try {
+            if (!(input.contains(startingFlag + " ") && input.contains(" " + endingFlag))) {
```
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java:39]: I think can abstract this if condition in another method for clarity

**Review by GlendonNotGlen — COMMENTED** [review]
Great job adding User class and updating the JUnit testing for each of the commands!

File: src/main/java/seedu/mindmymoney/helper/AddCommandInputTests.java [hunk 2]
```diff
@@ -114,4 +130,22 @@ public static void testAmount(String inputAmount) throws MindMyMoneyException {
         }
         assert inputAmountAsInteger > 0 : "Amount should have a positive value";
     }
+
+    public static void testIncomeAmount(int inputAmount) throws MindMyMoneyException {
+        if (inputAmount < 0) {
+            throw new MindMyMoneyException("Amount cannot be negative!");
+        } else if (inputAmount > 1000000) {  // Students typically would not have an income greater than 1,000,000
```
Comment (limjierui) [inline src/main/java/seedu/mindmymoney/helper/AddCommandInputTests.java:137]: If i remember correctly, i think the comment should be above instead of beside the code.

**Review by limjierui — COMMENTED** [review]
Overall good implementation of code for the User class. Followed the coding standards well too

**Review by SeanHoWB — COMMENTED** [review]
looks good so far, have to merge and test to see if got issues

File: src/main/java/seedu/mindmymoney/helper/AddCommandInputTests.java (hunk 2 above)
Comment (khseah) [reply src/main/java/seedu/mindmymoney/helper/AddCommandInputTests.java:137]: Ya good point, but in this case how to put above, cos it will intefere with the if else structure...
